src/core/lambda/io.rs

```diff
@@ -51,10 +51,12 @@ impl Eval for IO {
     ) -> Pin<Box<dyn Future<Output = Result<ConstValue, EvaluationError>> + 'a + Send>> {
         if ctx.request_ctx.upstream.dedupe {
             Box::pin(async move {
-                let key = self.cache_key(&ctx);
+                let key = self
+                    .cache_key(&ctx)
+                    .ok_or(anyhow::anyhow!("Unable to generate Cache Key"))?;
```

Comment:
I think this should not fail
it should skip deduping and log a warning
and call self.eval_inner(ctx)
@tusharmath Please suggest